scripts/SANS/sans/algorithm_detail/MoveSansInstrumentComponent.py

```diff
@@ -0,0 +1,122 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.py3compat.enum import Enum
+
+from sans.algorithm_detail.move_workspaces import create_mover
+from sans.common.enums import DetectorType
+
+
+class MoveTypes(Enum):
+    # Get which move operation the user wants to perform on the workspace. This can be:
+    # Initial move: Suitable when a workspace has been freshly loaded.
+    INITIAL_MOVE = 1
+    # Elementary displacement: Takes the degrees of freedom of the detector into account.
+    # This is normally used for beam center finding
+    ELEMENTARY_DISPLACEMENT = 2
+    # Set to zero: Set the component to its IDF position
+    RESET_POSITION = 3
+
+
+class MoveSansInstrumentComponent(object):
```

Comment:
Why is this a class?

From a quick `grep` it seems this functionality is used the class is created then immediately used, the fact it stores any data is redundant.
That fact that it is a class adds to the total quality of code and adds additional mental hurdles to understand what it is doing.

If something doesn't need to be a class, it shouldn't be.

Reply:
In this particular case it's a class as we can move multiple components (using the call) given relative beam_coordinates

Currently all callers assumed it was a workflow algorithm, so setup / call it in loops hence the create/discard. 
At a later date we can hoist the class creation out and share it instead, but that is beyond the scope of fixing workspace histories

Reply:
> At a later date we can hoist the class creation out and share it instead, but that is beyond the scope of fixing workspace histories

That will only lead to (more) difficult to understand code.
All this being a class does it allow you to wrap move type and beam coordinates in another structure instead of passing them to `move_component`.
I can't see why this is a better solution than just having a `move_component` function.

Reply:
Okay I'll go ahead and hoist this out and change the other workflow alg PRs similarly